pinot-core/src/main/java/org/apache/pinot/core/common/DataSource.java

```diff
@@ -60,6 +60,13 @@
   @Nullable
   InvertedIndexReader<?> getRangeIndex();
 
+  /**
+   * Returns the text index for the column if exists, or {@code null} if not.
+   * <p>TODO: Have a separate interface for text index.
+   */
+  @Nullable
+  InvertedIndexReader<?> getTextIndex();
```

Comment:
I'd like to do the cleanup for TextIndex in this PR itself. Both text and range followed the InvertedIndexReader interface.